src/pages/task/TaskTabs.tsx

```diff
@@ -164,9 +159,11 @@ export const TaskTabs: React.FC<TaskTabProps> = ({ task, taskFiles }) => {
     [TaskTab.TrendCharts]: isPerfPluginEnabled,
   };
 
-  const activeTabs = Object.keys(tabMap).filter(
-    (tab) => tabIsActive[tab]
-  ) as TaskTab[];
+  // showAnnotationsTab is the only thing that will change between renders since that is fetched after this component is already rendered
+  const activeTabs = useMemo(
```

Comment:
nit: i think this code is easier to understand this code without `useMemo` because then we don't have to think closely about when different data is fetched in the deps array and we don't have to disable exhuastive-deps